services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java

```diff
@@ -124,6 +128,21 @@ private KeyStore loadKeyStore(FileInputStream inputStream, String keystorePath)
         return keyStore;
     }
 
+    private KeyWrapper loadOctKey(RealmModel realm, ComponentModel model, KeyStore keyStore, String keyAlias, KeyUse use) throws UnrecoverableKeyException, KeyStoreException, NoSuchAlgorithmException {
+        SecretKey secretKey = (SecretKey) keyStore.getKey(keyAlias, model.get(JavaKeystoreKeyProviderFactory.KEY_PASSWORD_KEY).toCharArray());
```

Comment:
Maybe here we need to check the algorithm expected. Because we can have several types of secret keys. WDYT about checking the `secretKey.getAlgorithm()` to be the expected type (passed as another parameter to the `loadOctKey` method, in this case `AES`). If not correct, just throw an exception saying invalid algorithm for the key or similar.

Reply:
Indeed, facing some hard time here trying to load HMACSHA256 keys. No support from BCFKS, PKCS12, JKS.

There is no support for any other format in KC so far.